### PR TITLE
DEVX-1883: First kafka consumer application [Kafka 101 Series]

### PR DESCRIPTION
**Pull request by bbejeck**
Covers #443

### CODE REVIEW

**Comment by bbejeck** [comment]
[test for this PR](https://confluentinc.semaphoreci.com/jobs/d8257ffc-1914-472e-9a71-3a59720dfd0al) passed

**Comment by bbejeck** [comment]
PR ready for review

File: _includes/tutorials/kafka-consumer-application/kafka/markup/dev/create-topic.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+In this step we're going to create a topic for use during this tutorial.
+
+But your first, you're going to open a shell on the docker broker contaniner.
```
Comment (rspurgeon) [inline _includes/tutorials/kafka-consumer-application/kafka/markup/dev/create-topic.adoc:4]:
```suggestion
But first, you're going to open a shell on the broker docker container.
```

File: _includes/tutorials/kafka-consumer-application/kafka/markup/dev/make-dev-file.adoc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+`enable.auto.commit`:: Setting this configuration to `true` enables the Kafka consumer to handle commiting offsets automatically for you.  The default setting is `true`, but it's included here to make it explicit.  When you enable auto commit, you need to ensure you've processed all records _**before**_ the consumer calls `poll` again.  Once there is a subsequent call to `poll`, all the records returned from the previous call are considered processed and the consumer commits the offsets.
+
+`auto.offset.reset`::  If a consumer instance can't locate any offsets for its topic-partition assignment(s), it will resume processing from the _**earliest**_ available offset.
+
+`group.id`:: Kafka uses the concept of a consumer-group which is used to represent a logical single group.  A consumer-group can be made up of mulitple members all sharing the same `group.id` configuration.  As members leave or join the consumer-group, the group-coordinator triggres a rebalance which causes topic-partion reassignment among active members of the group.
+
```
Comment (rspurgeon) [inline _includes/tutorials/kafka-consumer-application/kafka/markup/dev/make-dev-file.adoc:18]: The mix of the Header (Deserializers) and the formatted text as header looks a little odd in my opinion.  Maybe you could remove the headers from each and just reference the properties in each paragraph.  Possibly add bullet points if that would look good.

File: _includes/tutorials/kafka-consumer-application/kafka/markup/dev/make-consumer-app.adoc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    this.consumer = consumer;
+    this.recordsHandler = recordsHandler;
+}
+----
+
+<1> Here you're suppling instances of the `Consumer` and `ConsumerRecordsHandler` via constructor parameters.
+
+By using interfaces vs. concreate implentations you can more easily test the `KafkaConsumerApplication` class by swapping in a `MockConsumer` for the test.  We'll cover testing in an upcoming section.  Also, interfaces make it simple to change `ConsumerRecord` handling at run-time.
+
+In this tutorial you'll inject the dependencies in the `KafkaConsumerApplication.main()` method, but in practice you'll may use some depnendency injection code such as the  https://spring.io/projects/spring-framework[Spring Framework].
```
Comment (rspurgeon) [inline _includes/tutorials/kafka-consumer-application/kafka/markup/dev/make-consumer-app.adoc:28]:
```suggestion
In this tutorial you'll inject the dependencies in the `KafkaConsumerApplication.main()` method, but in practice you'll may use some dependency injection code such as the  https://spring.io/projects/spring-framework[Spring Framework].
```

**Comment by rspurgeon** [comment]
@bbejeck did you reference the proper DEVX Jira issue for this?  DEVX-1881 is something unrelated.

File: _includes/tutorials/kafka-consumer-application/kafka/markup/dev/print-consumer-file-results.adoc [hunk 4]
```diff
@@ -0,0 +1,19 @@
+////
+  This is a sample content file for how to include a console consumer to the tutorial, probably a good idea so the end user can watch the results
+  of the tutorial.  Change the text as needed.
+
+////
+
+Your consumer application should have consumed all the records sent and written them out to a file.
+
+Run this command to print the results to the console:
```
Comment (rspurgeon) [inline _includes/tutorials/kafka-consumer-application/kafka/markup/dev/print-consumer-file-results.adoc:9]:
```suggestion
In a new terminal, run this command to print the results to the console:
```

**Comment by bbejeck** [comment]
rebased this PR

File: _includes/tutorials/kafka-consumer-application/kafka/markup/dev/create-topic.adoc (hunk 1 above)
Comment (bbejeck) [reply _includes/tutorials/kafka-consumer-application/kafka/markup/dev/create-topic.adoc:4]: ack

File: _includes/tutorials/kafka-consumer-application/kafka/markup/dev/make-dev-file.adoc (hunk 2 above)
Comment (bbejeck) [reply _includes/tutorials/kafka-consumer-application/kafka/markup/dev/make-dev-file.adoc:18]: Yep will do.  Good call.

File: _includes/tutorials/kafka-consumer-application/kafka/markup/dev/make-consumer-app.adoc (hunk 3 above)
Comment (bbejeck) [reply _includes/tutorials/kafka-consumer-application/kafka/markup/dev/make-consumer-app.adoc:28]: ack

**Comment by bbejeck** [comment]
Updated this PR per review comments.  The [test for this PR](https://confluentinc.semaphoreci.com/jobs/df70a724-f817-41db-aeb6-a32a7b14fd18) passed. 

Ping @rspurgeon for review of changes and a +1 for merging

**Comment by bbejeck** [comment]
rebased the PR

**Pull request merged**